metric_learn/base_metric.py

```diff
@@ -242,6 +242,118 @@ def transform(self, X):
     """
 
 
+class BilinearMixin(BaseMetricLearner, metaclass=ABCMeta):
+  r"""Bilinear similarity learning algorithms.
+
+  Algorithm that learns a Bilinear (pseudo) similarity :math:`s_M(x, x')`,
+  defined between two column vectors :math:`x` and :math:`x'` by: :math:
+  `s_M(x, x') =  x M x'`, where :math:`M` is a learned matrix. This matrix
+  is not guaranteed to be symmetric nor positive semi-definite (PSD). Thus
+  it cannot be seen as learning a linear transformation of the original
+  space like Mahalanobis learning algorithms.
+
+  Attributes
+  ----------
+  components_ : `numpy.ndarray`, shape=(n_components, n_features)
```

Comment:
Two important comments:
- The shape is incorrect: it will always be (n_features, n_features)
- I am not sure we should name this `components_` because this is typically used to denote a projection matrix used to transform the data (as in PCA, or Mahalanobis learners). I think `bilinear_matrix_` would be more appropriate. Not sure if having a different name would be at odds with the generality of some tests?